tsn/src/lib.rs

```diff
@@ -0,0 +1,346 @@
+use std::io::Error;
+use std::{mem, str};
+
+extern crate socket;
+
+#[derive(Clone, Copy)]
+struct TsnSocket {
+    fd: i32,
+    ifname: [u8; 40],
+    vlanid: u32,
+}
+
+static mut SOCKETS: [TsnSocket; 20] = [TsnSocket {
+    fd: 0,
+    ifname: [0; 40],
+    vlanid: 0,
+}; 20];
+
+const CONTROL_SOCK_PATH: &str = "/var/run/tsn.sock\x00";
```

Comment:
`String` can be automatically converted to `&str`

Reply:
Compile errors about not providing a type for the constant.
```
error: missing type for `const` item
  --> src/lib.rs:21:7
   |
21 | const CONTROL_SOCK_PATH = "/var/run/tsn.sock\x00";
   |       ^^^^^^^^^^^^^^^^^ help: provide a type for the constant: `CONTROL_SOCK_PATH: &str`

error: could not compile `tsn` due to previous error
```